a10_octavia/controller/worker/flows/a10_listener_flows.py

```diff
@@ -33,21 +35,56 @@ def get_create_listener_flow(self):
-        create_listener_flow.add(lifecycle_tasks.ListenersToErrorOnRevertTask(
-            requires=[constants.LOADBALANCER, constants.LISTENERS]))
-        create_listener_flow.add(a10_database_tasks.GetVThunderByLoadBalancer(
-            requires=constants.LOADBALANCER,
-            provides=a10constants.VTHUNDER))
-        create_listener_flow.add(virtual_port_tasks.ListenersCreate(
-            requires=[constants.LOADBALANCER, constants.LISTENERS, a10constants.VTHUNDER]))
+        create_listener_flow.add(*self._get_create_listener_subflow())
         create_listener_flow.add(a10_network_tasks.UpdateVIP(
             requires=constants.LOADBALANCER))
-        create_listener_flow.add(database_tasks.
-                                 MarkLBAndListenersActiveInDB(
+        create_listener_flow.add(a10_database_tasks.
+                                 MarkLBAndListenerActiveInDB(
                                      requires=[constants.LOADBALANCER,
-                                               constants.LISTENERS]))
+                                               constants.LISTENER]))
         return create_listener_flow
 
+    def _get_create_listener_subflow(self):
+        create_listener_flow = linear_flow.Flow(a10constants.CREATE_LISTENER_SUBFLOW)
+        create_listener_flow.add(lifecycle_tasks.ListenerToErrorOnRevertTask(
+            requires=constants.LISTENER))
+        create_listener_flow.add(a10_database_tasks.GetVThunderByLoadBalancer(
+            requires=constants.LOADBALANCER,
+            provides=a10constants.VTHUNDER))
+        create_listener_flow.add(*self.get_listener_create_cert_template_subflow())
+        create_listener_flow.add(virtual_port_tasks.ListenerCreate(
+            requires=[constants.LOADBALANCER, constants.LISTENER, a10constants.VTHUNDER]))
+        return create_listener_flow
+
+    def _get_cert_create_subflow(self):
+        get_create_cert_subflow = linear_flow.Flow(a10constants.CREATE_CERT_TEMPLATE_SUBFLOW)
+        get_create_cert_subflow.add(cert_tasks.SSLKeyCreate(
+            requires=(a10constants.CERT_DATA, a10constants.VTHUNDER)))
+        get_create_cert_subflow.add(cert_tasks.ClientSSLTemplateCreate(
+            requires=(a10constants.CERT_DATA, a10constants.VTHUNDER)))
+        return get_create_cert_subflow
+
+    def get_listener_create_cert_template_subflow(self):
+        get_listener_create_cert_subflow = graph_flow.Flow(
+            a10constants.CREATE_LISTENER_CERT_SUBFLOW)
+        cert_create_task = cert_tasks.SSLCertCreate(
+            requires=[constants.LOADBALANCER, constants.LISTENER, a10constants.VTHUNDER],
+            provides=a10constants.CERT_DATA)
+        cert_create_template_task = self._get_cert_create_subflow()
+        get_listener_create_cert_subflow.add(cert_create_task, cert_create_template_task)
+        get_listener_create_cert_subflow.link(
+            cert_create_task, cert_create_template_task, decider=self._is_terminated_https)
```

Comment:
@sridharsrinivas 
We are using `SSLCertCreate` task for decider purpose in task flow to avoid an extra task addition and reuse what we already have.

However as far as `SSLCertCreate` is concerned, it won't get executed in non-https condition at all because in its definition we placed a condition saying `if listener.protocol == "TERMINATED_HTTPS"`.